src/services/ReportHub/ReportHub.WebApi/Controllers/ProjectsInfoController.cs

```diff
@@ -0,0 +1,38 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using ReportHub.BLL.Contracts;
+
+namespace ReportHub.WebApi.Controllers;
+
+/// <summary>
+/// Controller for managing projects info.
+/// </summary>
+[ApiController]
+[Authorize]
+[Route("api/[controller]")]
+public class ProjectsInfoController : ControllerBase
+{
+    private readonly IProjectInfoService _projectInfoService;
+
+    /// <summary>
+    /// Constructor for ProjectsInfoController.
+    /// </summary>
+    public ProjectsInfoController(IProjectInfoService projectInfoService)
+    {
+        _projectInfoService = projectInfoService;
+    }
+
+    /// <summary>
+    /// Get all awailable user project infos
+    /// </summary>
+    [HttpGet("all")]
+    public async Task<IActionResult> GetAllUserProjectInfos(
```

Comment:
1)Fix the routing
2)For asynchronous methods, it's necessary to append Async to the end of the method name